plugin/trino-druid/src/test/java/io/trino/plugin/druid/TestDruidCaseInsensitiveMatch.java

```diff
@@ -0,0 +1,90 @@
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test(singleThreaded = true)
+public class TestDruidCaseInsensitiveMatch
+        extends AbstractTestQueryFramework
+{
+    private TestingDruidServer druidServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        druidServer = new TestingDruidServer();
+        closeAfterClass(() -> {
+            druidServer.close();
+            druidServer = null;
+        });
+        DistributedQueryRunner queryRunner = DruidQueryRunner.createDruidQueryRunnerTpch(
+                druidServer, ImmutableMap.of(), ImmutableMap.of("case-insensitive-name-matching", "true"));
+        copyAndIngestTpchData(queryRunner.execute(SELECT_FROM_ORDERS + " LIMIT 10"), this.druidServer, "orders", "CamelCase");
+        return queryRunner;
+    }
+
+    @Test
+    public void testNonLowerCaseTableName()
+    {
+        MaterializedResult expectedColumns = MaterializedResult.resultBuilder(getQueryRunner().getDefaultSession(), VARCHAR, VARCHAR, VARCHAR, VARCHAR)
+                .row("__time", "timestamp(3)", "", "")
+                .row("clerk", "varchar", "", "") // String columns are reported only as varchar
+                .row("comment", "varchar", "", "")
+                .row("custkey", "bigint", "", "") // Long columns are reported as bigint
+                .row("orderdate", "varchar", "", "")
+                .row("orderkey", "bigint", "", "")
+                .row("orderpriority", "varchar", "", "")
+                .row("orderstatus", "varchar", "", "")
+                .row("shippriority", "bigint", "", "") // Druid doesn't support int type
+                .row("totalprice", "double", "", "")
+                .build();
+        MaterializedResult actualColumns = computeActual("DESCRIBE " + "CamelCase");
+        Assert.assertEquals(actualColumns, expectedColumns);
+        MaterializedResult materializedRows = computeActual("SELECT * FROM druid.druid.CAMELCASE");
+        Assert.assertEquals(materializedRows.getRowCount(), 10);
+        MaterializedResult materializedRows1 = computeActual("SELECT * FROM druid.CamelCase");
+        MaterializedResult materializedRows2 = computeActual("SELECT * FROM druid.camelcase");
+        assertThat(materializedRows.equals(materializedRows1));
+        assertThat(materializedRows.equals(materializedRows2));
+    }
+
+    @Test
+    public void testTableNameClash()
+            throws IOException, InterruptedException
+    {
+        try {
+            //ingesting data with already existing table name in lowercase which should fail
+            copyAndIngestTpchData(getQueryRunner().execute(SELECT_FROM_REGION + " LIMIT 10"), this.druidServer, "region", "camelcase");
+        }
+        catch (AssertionError e) {
+            Assert.assertEquals(e.getMessage(), "Datasource camelcase not loaded expected [true] but found [false]");
```

Comment:
Can you share the entire stack trace?

This exception itself can also happen if something fails on Druid end rather than our side.